macro/src/codegen.rs

```diff
@@ -267,20 +269,23 @@ impl SpannedClassInfo {
                             self,
                             jvm: &mut Jvm<'jvm>,
                             (): (),
-                        ) -> duchess::Result<Self::Output<'jvm>> {
+                        ) -> duchess::Result<'jvm, Self::Output<'jvm>> {
                             #(#prepare_inputs)*
 
-                            let class = cached_class(jvm)?;
+                            let class = <#ty>::class(jvm)?;
+                            // XX: safety
+                            let jclass = unsafe { JClass::from_raw(class.as_jobject().as_raw()) };
```

Comment:
Realized this after the fact, but there's a built-in `Into` that can do this without an unsafe. Will fix as part of the next PR